### PR TITLE
Refactory bits for the codegen cache

### PR DESCRIPTION
**Pull request by ani300**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #103453
* __->__ #103452

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @ngimel @yf225 @chenyang78 @aakhundov

### CODE REVIEW

**Comment by ani300** [comment]
Please see #101027 for previous discussion on this PR

File: torch/_inductor/compile_fx.py [hunk 1]
```diff
@@ -424,6 +384,97 @@ def compiled_fn(new_inputs):
     return result
 
 
+def fx_codegen_and_compile(
```
Comment (ezyang) [inline torch/_inductor/compile_fx.py:387]: What's going on here?

Comment (ani300) [reply torch/_inductor/compile_fx.py:387]: I split the `inner_compile_fx` function into 2: `fx_codegen_and_compile` and `inner_fx_compile`. The first function is the one that will be cached by the cache implemented in #103453, while the rest of `inner_fx_compile` contains all the cudagrahs code that can't be as easily cached due to the runtime requirement.

Comment (ani300) [reply torch/_inductor/compile_fx.py:387]: Then I return the new object `CompiledFxGraph`, which is the one that can be cached. If you have a better name for the function splits, I'm happy to take suggestions.

**Review by ezyang — APPROVED** [review]

File: torch/_inductor/compile_fx.py [hunk 2]
```diff
@@ -448,15 +499,19 @@ def is_aligned(storage_offset, dtype):
...
 
     def run(new_inputs):
         for i in check_inputs:
             if new_inputs[i].data_ptr() % ALIGNMENT:
                 new_inputs[i] = clone_preserve_strides(new_inputs[i])
-        return model(new_inputs)
+        return old_compiled_artifact(new_inputs)
 
-    return run
+    compiled_graph.current_callable = run
```
Comment (eellison) [inline torch/_inductor/compile_fx.py:512]: Why is updating the `current_callable` necessary ?

**Pull request closed without merging**